packages/bot-web-ui/src/stores/journal-store.js

```diff
@@ -76,14 +80,20 @@ export default class JournalStore {
         const unique_id = Blockly.utils.genUid();
 
         this.unfiltered_messages.unshift({ date, time, message, message_type, className, unique_id, extra });
+        localStorage.setItem(journal_storage_key, JSON.stringify(this.unfiltered_messages));
```

Comment:
Could we use a Mobx `reaction` for this to separate updating state + a side effect? (can we apply this to all places where we're writing to local storage)